evalml/pipelines/components/transformers/encoders/onehot_encoder.py

```diff
@@ -180,10 +178,38 @@ def categories(self, feature_name):
             raise ValueError(f'Feature "{feature_name}" was not provided to one-hot encoder as a training feature')
         return self._encoder.categories_[index]
 
+    @staticmethod
+    def _make_name_unique(name, seen_before):
+        """Helper to make the name unique."""
+        i = 1
+        name = f"{name}_{i}"
+        while name in seen_before:
+            name = f"{name[:name.rindex('_')]}_{i}"
+            i += 1
+        return name
+
     def get_feature_names(self):
         """Return feature names for the input features after fitting.
 
         Returns:
             np.array: The feature names after encoding, provided in the same order as input_features.
         """
-        return self._encoder.get_feature_names(self.features_to_encode)
+        unique_names = []
+        seen_before = set([])
+        for col_index, col in enumerate(self.features_to_encode):
+            column_categories = self.categories(col)
+            for cat_index, category in enumerate(column_categories):
+
+                # Drop categories specified by the user
+                if self._encoder.drop_idx_ is not None and self._encoder.drop_idx_[col_index] is not None:
+                    if cat_index == self._encoder.drop_idx_[col_index]:
+                        continue
+
+                # Follow sklearn naming convention but if name has been seen before
+                # then add an int to make it unique
+                proposed_name = f"{col}_{category}"
+                if proposed_name in seen_before:
+                    proposed_name = self._make_name_unique(proposed_name, seen_before)
+                unique_names.append(proposed_name)
+                seen_before.add(proposed_name)
```

Comment:
Is it necessary to have these be two separate variables? It looks like they store the same information

Reply:
Good question! The `_make_name_unique` needs to repeatedly check whether we have already seen a name so a set is better than a list for that I think. The other option is to create a set from `unique_names` inside `_make_name_unique` but I figured we might as well just keep one set around at that point. What do you think? 

We definitely need to return a list because the order of the columns matters.

Reply:
I see, that makes a lot of sense. Totally fine with keeping both!